[Workers] Source Map beta (#13815)
cloudflare/cloudflare-docs: content/workers/observability/source-maps.md

---
pcx_content_type: concept
title: Source maps and stack traces
meta:
  description: Adding source maps and generating stack traces for Workers. 
---
{{<heading-pill style="beta">}}Source maps and stack traces{{</heading-pill>}}

> [Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. They show you the specific functions that were called, in what order, from which line and file, and with what arguments.

Reviewer: ```suggestion
[Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. Stack traces show you the specific functions that were called, in what order, from which line and file, and with what arguments.
```

> [Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. They show you the specific functions that were called, in what order, from which line and file, and with what arguments.

Reviewer: ```suggestion
[Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. Stack traces show you the specific functions that were called, in what order, from which line and file, and with what arguments.
```

> Most JavaScript code – not just on Workers, but across platforms – is first bundled, often transpiled, and then minified before being deployed to production. This to create smaller bundles to optimize performance and convert from Typescript to Javascript if needed.

Reviewer: ```suggestion
Most JavaScript code – not just on Workers, but across platforms – is first bundled, often transpiled, and then minified before being deployed to production. This process creates smaller bundles to optimize performance and converts code from Typescript to Javascript if needed.
```

> Source maps translate compiled and minified code back to the original code that you wrote. Source maps are combined with the stack trace returned by the JavaScript runtime in order to present you with a stack trace.

Reviewer: ```suggestion
Source maps translate compiled and minified code back to the original code that you wrote. Source maps are combined with the stack trace returned by the JavaScript runtime to present you with a stack trace.
```

{{<Aside type="warning">}}

> Source maps and stack traces support is going to be in open beta on April 15. 

Reviewer 2: ```suggestion
Source maps and stack traces support will be available in open beta starting on April 15. 
```

{{</Aside>}}

## Source Maps

> Source maps can be enabled on a per-Worker level by adding the following to the Worker's [wrangler.toml](/workers/wrangler/) file: 

Reviewer 2: ```suggestion
Add the following to your Worker's [wrangler.toml](/workers/wrangler/) file to enable source maps: 
```

> Source maps can be enabled on a per-Worker level by adding the following to the Worker's [wrangler.toml](/workers/wrangler/) file: 

Reviewer: ```suggestion
To enable source maps, add the following to your Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file: 
```

```toml
upload_source_maps = true 

> 

Reviewer: ```suggestion
```

```

> Once this setting is configured, wrangler will automatically generate and upload any source map files when you run [`wrangler deploy`](/workers/wrangler/commands/#deploy) or [`wrangler versions deploy`](/workers/wrangler/commands/#deploy-2). 

Reviewer 2: ```suggestion
When `upload_source_maps` is set to `true`, Wrangler will automatically generate and upload source map files when you run [`wrangler deploy`](/workers/wrangler/commands/#deploy) or [`wrangler versions deploy`](/workers/wrangler/commands/#deploy-2).
```

​​

{{<Aside type="note">}}

> Miniflare supports [uploading source maps for local development](https://miniflare.dev/developing/source-maps). 

Reviewer 2: ```suggestion
Miniflare can also [output source maps](https://miniflare.dev/developing/source-maps) for use local development or [testing](/workers/testing/integration-testing/#miniflares-api). 
```

{{</Aside>}}

> ## Stack Traces

Reviewer: ```suggestion
## Stack traces
```

​​

> When your Worker throws an uncaught exception, we fetch the source map and use it to map the stack trace of the exception back to lines of your Worker’s original source code.

Reviewer 2: ```suggestion
When your Worker throws an uncaught exception, after your Worker invocation completes, the Workers runtime will fetch the source map file that you uploaded and use it to map the stack trace of the exception back to lines of your Worker’s original source code.
```

Can rephrase if you want but think important to be clear that you won't get a sourcemapped stack trace for an error that is exposed at runtime in your own Worker

Author: Yep, agreed. Adding
